app/models/order.rb

```diff
@@ -6,10 +6,41 @@ class Order < ActiveRecord::Base
   has_many :shipments
 
   # Order processing flowchart
-  # pending -> approved -> filled -> shipped -> received -> closed
-  #        `-> rejected
+  # select_items -> select_ship_to -> confirm_order -/
+  # ,----------------------------------------------~'
+  # `-> pending -> approved -> filled -> shipped -> received -> closed
+  #            `-> rejected
+
+  enum status: { select_items: -3,
+                 select_ship_to: -2,
+                 confirm_order: -1,
+                 pending: 0,
+                 approved: 1,
+                 rejected: 2,
+                 filled: 3,
+                 shipped: 4,
+                 received: 5,
+                 closed: 6 } do
+    event :choose_items do
+      transition select_items: :select_ship_to
+    end
+
+    event :edit_items do
+      transition [:select_ship_to, :confirm_order] => :select_items
+    end
+
+    event :edit_ship_to do
+      transition confirm_order: :select_ship_to
+    end
+
+    event :choose_ship_to do
+      transition select_ship_to: :confirm_order
+    end
+
+    event :submit_order do
```

Comment:
if we ever extended other functionality that needed a submit action (submit_tracking_info) I think it would be nice to be declarative about what is being submitted.